culprit_finder/src/culprit_finder/github.py

```diff
@@ -0,0 +1,195 @@
+"""
+Module for interacting with the GitHub API via the gh CLI.
+"""
+
+import subprocess
+import json
+import logging
+from typing import Optional, TypedDict
+
+
+class CommitDetail(TypedDict):
+  message: str
+
+
+class Commit(TypedDict):
+  sha: str
+  commit: CommitDetail
+
+
+class Workflow(TypedDict):
+  name: str
+  path: str
+
+
+class Run(TypedDict):
+  headSha: str
+  status: str
+  createdAt: str
+  conclusion: Optional[str]
+  databaseId: int
+  url: str
```

Comment:
Could we add a docstring to this TypedDict to clarify the fields? Most here are self-explanatory but documenting specific ones like `databaseId` would be useful.